[Bug] Issue with Long dependency being listed as devDependency

### What are you really trying to do?

Trying to get worker deployments to work correctly in a container that uses yarn to only install dependencies, not devDependencies.

### Describe the bug

Running the workers in Docker containers that are built with `yarn workspaces focus --production @packages/temporal-workflows` results in workers that won't execute with the following stack trace:
```
2022-08-04T03:32:56.766Z [ERROR] Failed to activate workflow {
  runId: 'b2fd073b-42a6-40ce-b719-1468474d7f2c',
  error: TypeError: (seconds || long_1.default.UZERO).mul is not a function
      at tsToMs (/opt/app-root/src/.yarn/cache/@temporalio-internal-workflow-common-npm-1.0.0-rc.1-b66f54d7d6-4272b669aa.zip/node_modules/@temporalio/internal-workflow-common/lib/time.js:30:10)
      at /opt/app-root/src/.yarn/cache/@temporalio-worker-npm-1.0.0-rc.1-2998f0d7a7-147819cf17.zip/node_modules/@temporalio/worker/lib/worker.js:664:90
      at async /opt/app-root/src/.yarn/cache/@temporalio-worker-npm-1.0.0-rc.1-2998f0d7a7-147819cf17.zip/node_modules/@temporalio/worker/lib/tracing.js:65:20
      at async /opt/app-root/src/.yarn/cache/@temporalio-worker-npm-1.0.0-rc.1-2998f0d7a7-147819cf17.zip/node_modules/@temporalio/worker/lib/worker.js:593:28,
  workflowExists: false
}
```

However, if we build this instead with `yarn workspaces focus --all @packages/temporal-workflows`, the worker/container works, which implies that some devDependency needs to be listed as a dependency (likely the `long` package).

### Minimal Reproduction

I'm having issues figuring out how to reproduce this or test this to be honest. I tried testing this solution by forking this repository to see if making the suggested changes would fix it but run into errors trying to use the forked repository as my `temporalio` version. I get the following error (https://community.temporal.io/t/fork-temporal-sdk-typescript/5630).
<!-- 
Modify our hello world templates to demonstrate:

- TypeScript: https://github.com/temporalio/samples-typescript/tree/main/hello-world
- Go: https://github.com/temporalio/money-transfer-project-template-go
- Java: https://github.com/temporalio/money-transfer-project-template-java
- PHP: https://github.com/temporalio/samples-php#samples
-->

### Environment/Versions

<!-- Please complete the following information where relevant. -->
Containers have been built on both M1Macs and Linux machines, using the latest Temporal SDK version. We're using yarn pnp and 3.2.0.

### Additional context

Related to [this forum post](https://community.temporal.io/t/worker-error-with-mul-is-not-a-function/5573)

--- a/packages/proto/protos/root.js
+++ b/packages/proto/protos/root.js
@@ -1,3 +1,9 @@
+// Workaround an issue that prevents protobufjs from loading 'long' in Yarn 3 PnP
+// https://github.com/protobufjs/protobuf.js/issues/1745#issuecomment-1200319399
+const $protobuf = require('protobufjs/light');
+$protobuf.util.Long = require('long');
+$protobuf.configure();
+
 const { patchProtobufRoot } = require('../lib/patch-protobuf-root');
 const unpatchedRoot = require('./json-module');
 module.exports = patchProtobufRoot(unpatchedRoot);

--- a/packages/testing/scripts/compile-proto.mjs
+++ b/packages/testing/scripts/compile-proto.mjs
@@ -47,6 +47,20 @@ async function compileProtos(protoPath, jsOutputFile, dtsOutputFile, ...args) {
   ];
   await promisify(pbjs.main)(pbjsArgs);
 
+  // Workaround an issue that prevents protobufjs from loading 'long' in Yarn 3 PnP
+  // https://github.com/protobufjs/protobuf.js/issues/1745#issuecomment-1200319399
+  const pbjsOutput = readFileSync(jsOutputFile, 'utf8');
+  writeFileSync(
+    jsOutputFile,
+    pbjsOutput.replace(
+      /(require\("protobufjs\/minimal"\);)$/m,
+      `$1
+       $protobuf.util.Long = require('long');
+       $protobuf.configure();
+    `
+    )
+  );
+
   console.log(`Creating protobuf TS definitions from ${protoPath}`);
   await promisify(pbts.main)(['--out', dtsOutputFile, jsOutputFile]);
 

